### ISSUE TITLE
DOM selectors for source pane

### ISSUE THREAD

**Issue by trestletech** [rstudio/rstudio#6817]
We'd like to use `rstudioapi::higlightUI` to select some elements inside the source window. We're hoping to be able to highlight the following areas

### 1. The whole source pane

![highlight 001](https://user-images.githubusercontent.com/1593639/81190019-d048da80-8fa6-11ea-9f43-77497b367927.png)

### 2. Maximize/minimize buttons on the source pane

![highlights 021](https://user-images.githubusercontent.com/1593639/81191139-36822d00-8fa8-11ea-870f-96c8a2823737.png)

### ~~3. Buttons at the top of an RMD file~~

~~Each of the icons at the top of the Rmd file (only one illustrated here). In each case, it appears that there is a DOM ID associated with each of the items in the drop down list associated with the icon, but not a DOM ID associated with the icon itself.~~

EDIT: will move to a separate issue

![highlights 020](https://user-images.githubusercontent.com/1593639/81191290-603b5400-8fa8-11ea-94aa-44c3d3ec7eb2.png)

### 4. The outline section of an RMD doc

![highlights 022](https://user-images.githubusercontent.com/1593639/81191363-79440500-8fa8-11ea-8e36-ec58408834a4.png)

**Comment by gtritchie** [comment]
If low-risk I will put into 1.3-patch, otherwise will have to wait until 1.4.

**Comment by gtritchie** [comment]
Taking out of 1.3 patch. Requires touching a lot more code than is justified in 1.3-patch.

**The minimize-maximize buttons, highlighting the whole source pane, and the outline pane, are relatively easy, so if you want to narrow this issue down to those then can get into 1.3-patch.**

It's the editor toolbar buttons that are a nightmare. The goal is to give them unique classes starting with `rstudio_` which causes them to show up via the Help / Diagnostics / Show DOM tool. This avoids the problem with using the IDs which have to be kept unique (and thus end up with _## appended to them making them variable). Classes don't have that limit.

Unfortunately the editor buttons are created in a wide variety of tricky ways! So doing that comprehensively needs to wait for 1.4.

**Comment by trestletech** [comment]
Sure! We'll take whatever we can get. I just updated the description above to reflect that we can pull #3 out into a separate issue.

If you were able to do some or all of the others in 1.3, I'd be thrilled. Thanks!

**Comment by gtritchie** [comment]
Fixed by #6832

**Comment by R-Hannibal** [comment]
@trestletech - We should have items 1, 2, and 4 from the original request fulfilled in the latest 1.3 patch preview build (available [HERE](https://rstudio.com/products/rstudio/download/preview/)).

Do you mind confirming that those are looking like they've been successfully implemented for you in that build? Absolutely please let me know if that's inconvenient.

Also, any chance you have the issue number from item 3, or does that still need to be entered?

**Comment by trestletech** [comment]
Looks good!
